app/policies/account_applications_policy.rb

```diff
@@ -2,4 +2,13 @@ class AccountApplicationsPolicy < BasePolicy
   def index?
     current_user.govuk_admin?
   end
+
+  alias_method :show?, :index?
+  alias_method :grant_signin_permission?, :index?
+
+  def remove_signin_permission?
+    current_user.govuk_admin? && current_user.has_access_to?(record)
+  end
+
+  alias_method :view_permissions?, :remove_signin_permission?
```

Comment:
Thought: I know it's not how other policies have been implemented, but I wonder if this code would be more readable if we removed the duplication in a different way e.g. something like:

```ruby
  def index? = current_user_is_govuk_admin?
  def show? = current_user_is_govuk_admin?
  def grant_signin_permission? = current_user_is_govuk_admin?

  def remove_signin_permission? = current_user_is_govuk_admin? && current_user_has_access_to_application?
  def view_permissions? = current_user_is_govuk_admin? && current_user_has_access_to_application?

private

  def application = record
  def current_user_is_govuk_admin? = current_user.govuk_admin?
  def current_user_has_access_to_application? = current_user.govuk_admin? && current_user.has_access_to?(application)
```

🤷

Reply:
As we've discussed, I think all of the policy code could do with a bit of an overhaul, but we probably need to decide on a preferred approach before we do that. So I think this is definitely out-of-scope for this PR.